Tests/L0/SonarQubePostTest/PRCA/PostCommentsTests.ps1

```diff
@@ -125,26 +219,26 @@ function BuildTestComment
 #
 function ValidateDiscussionThreadCollection
 {
-    param ([Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threadCollection, [Array]$inputComments)
+    param ([Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threadCollection, [Array]$messages, [string]$commentSource)
     
     $endLineName = [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadPropertyNames]::EndLine
     $startLineName = [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadPropertyNames]::StartLine
     $itemPathName = [Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadPropertyNames]::ItemPath
     
-    ValidateCommonThreadAttributes $threadCollection
-    Assert-AreEqual $inputComments.Count $threadCollection.Count "Inconsistent number of threads. There should be 1 thread for each comment."    
-    foreach ($inputComment in $inputComments)
+    ValidateCommonThreadAttributes $threadCollection $commentSource
+    Assert-AreEqual $messages.Count $threadCollection.Count "Inconsistent number of threads. There should be 1 thread for each comment."    
+    
+    foreach ($message in $messages)
     {
         $thread = $threadCollection | Where-Object {
-            ($_.Comments[0].Content -eq $inputComment.Content) -and 
-            ($_.Properties[$startLineName] -eq $inputComment.Line) }
+            ($_.Comments[0].Content -eq $message.Content) -and 
+            ($_.Properties[$itemPathName] -eq $message.RelativePath) }
             
-        Assert-IsNotNullOrEmpty $thread "Could not find a thread associated with the comment on line $($inputComment.Line) with message $($inputComment.Content)"
-        Assert-AreEqual $inputComment.RelativePath $thread.Properties[$itemPathName] "Invalid ItemPath property"
-        Assert-AreEqual $inputComment.Line $thread.Properties[$endLineName] "The EndLine should be the same as the comment's line"
+        Assert-IsNotNullOrEmpty $thread "Could not find a thread associated with the comment on line $($message.Line) with message $($message.Content)"
+        Assert-AreEqual 1 $thread.Count "A single thread associated with this message should have been found"
```

Comment:
Where-Object will return an array. However if the array has a single element, it will be returned. This is a general PowerShell behaviour. However .Count seems to work consistently